File: Trackster/Trackster.Model/Media.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;

namespace Trackster.Model
{
    public class Media
    {
        [Key]
        public int media_id { get; set; }
        public string title { get; set; }
        public string description { get; set; }
        public DateTime release_date { get; set; }
        public string status { get; set; }
        public float rating { get; set; }
        public string? picture { get; set; }
        public string? backdrop { get; set; }

        [ForeignKey("language_id")]
        public virtual Languages language { get; set; }
        public int language_id { get; set; }

        [ForeignKey("streaming_service_id")]
        public virtual StreamingServices streaming_service { get; set; }
        public int streaming_service_id { get; set; }
        public string trailerUrl { get; set; }
        public bool? state { get; set; }
        public string? state_machine { get; set; }
    }
}

File: UI/trackster_mobile/lib/models/media.dart
import 'dart:ffi';

import 'package:trackster_mobile/models/genre.dart';
import 'package:trackster_mobile/models/language.dart';
import 'package:trackster_mobile/models/streaming_service.dart';
import 'package:json_annotation/json_annotation.dart';

part 'media.g.dart';

@JsonSerializable()
class Media {
  int? media_id;
  String? title;
  String? description;
  DateTime? release_date;
  String? status;
  double? rating;
  String? picture;
  String? backdrop;
  Language? language;
  StreamingService? streaming_service;
  bool? state;
  String? state_machine;

  Media({
    this.media_id,
    this.title,
    this.description,
    this.release_date,
    this.status,
    this.rating,
    this.picture,
    this.backdrop,
    this.language,
    this.streaming_service,
    this.state,
    this.state_machine,
  });

  factory Media.fromJson(Map<String, dynamic> json) => _$MediaFromJson(json);
  Map<String, dynamic> toJson() => _$MediaToJson(this);

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is Media &&
          runtimeType == other.runtimeType &&
          media_id == other.media_id;

  @override
  int get hashCode => media_id.hashCode;
}
